src/SAWScript/Prover/Versions.hs

```diff
@@ -0,0 +1,63 @@
+module SAWScript.Prover.Versions where
+
+import Data.List (intercalate)
+import Data.Maybe (catMaybes)
+import System.Directory (findExecutable)
+import System.Process (readProcess)
+import Text.Read (readMaybe)
+
+data Version
+  = Version [Int]
+  deriving (Eq, Ord, Show)
+
+parseVersion :: String -> Version
+parseVersion = Version . reverse . catMaybes . map readMaybe . go [[]]
+  where
+    go (n : ns) (c : cs) | '0' <= c && c <= '9' = go ((c : n) : ns) cs
```

Comment:
Doesn't this reverse the list of digits?

Reply:
Since it iteratively adds digits to the head of the accumulator passed into `go`, the digits start out reversed and need to be flipped around when it's done. I got this wrong when I first wrote it, but this version checks out in practice.

Reply:
I can see that the components of the version are reversed... but don't digits _within_ each component need to be reversed also?

Reply:
Oh, you're right! All the versions I tested it on had single-digit version components.